api_moq_test.go

```diff
@@ -335,3 +347,34 @@ func (mock *APIMock) SearchCheckBundlesCalls() []struct {
 	mock.lockSearchCheckBundles.RUnlock()
 	return calls
 }
+
+// UpdateCheckBundle calls UpdateCheckBundleFunc.
+func (mock *APIMock) UpdateCheckBundle(cfg *apiclient.CheckBundle) (*apiclient.CheckBundle, error) {
+	if mock.UpdateCheckBundleFunc == nil {
+		panic("APIMock.UpdateCheckBundleFunc: method is nil but API.UpdateCheckBundle was just called")
+	}
+	callInfo := struct {
+		Cfg *apiclient.CheckBundle
+	}{
+		Cfg: cfg,
+	}
+	mock.lockUpdateCheckBundle.Lock()
+	mock.calls.UpdateCheckBundle = append(mock.calls.UpdateCheckBundle, callInfo)
+	mock.lockUpdateCheckBundle.Unlock()
+	return mock.UpdateCheckBundleFunc(cfg)
+}
+
+// UpdateCheckBundleCalls gets all the calls that were made to UpdateCheckBundle.
+// Check the length with:
+//     len(mockedAPI.UpdateCheckBundleCalls())
+func (mock *APIMock) UpdateCheckBundleCalls() []struct {
+	Cfg *apiclient.CheckBundle
+} {
+	var calls []struct {
+		Cfg *apiclient.CheckBundle
+	}
+	mock.lockUpdateCheckBundle.RLock()
+	calls = mock.calls.UpdateCheckBundle
```

Comment:
Do you need to declare calls separately? Could this line just be `calls := mock.calls.UpdateCheckBundle`?

Reply:
the mock funcs are defined in the context of each test, since the test dictates what the func returns. e.g. 

```
client: &APIMock{
				UpdateCheckBundleFunc: func(cfg *apiclient.CheckBundle) (*apiclient.CheckBundle, error) {
					return cfg, nil
				},
			},
```